File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution(object):
    def longestConsecutive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        seq=set()
        for num in nums:
            seq.add(num)
        max1=0
        count=0
        for num in nums:
            if (num-1) not in seq:
                start=num
                while num in seq:
                    num+=1
                max1=max(max1,num-start)
        return max1

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public int longestConsecutive(int[] nums) {
        // Create a HashSet and add all elements from nums
        Set<Integer> numberSet = new HashSet<>();
        for (int num : nums) {
            numberSet.add(num);
        }

        int maxLen = 0;

        // Iterate through each number in the array
        for (int num : nums) {
            // Check if it's the start of a sequence
            if (!numberSet.contains(num - 1)) {
                int currentNum = num;
                int currentLen = 1;

                // Find the length of the sequence
                while (numberSet.contains(currentNum + 1)) {
                    currentNum++;
                    currentLen++;
                }

                // Update maxLen if necessary
                maxLen = Math.max(maxLen, currentLen);
            }
        }

        return maxLen;
    }
}
